mlrun/datastore/datastore.py

```diff
@@ -138,11 +138,16 @@ def secret(self, key):
     def _add_store(self, store):
         self._stores[store.name] = store
 
-    def get_store_artifact(self, url, project="", allow_empty_resources=None):
+    def get_store_artifact(
+        self, url, project="", allow_empty_resources=None, secrets=None
+    ):
 
         try:
             resource = get_store_resource(
-                url, db=self._get_db(), secrets=self._secrets, project=project
+                url,
+                db=self._get_db(),
+                secrets=secrets or self._secrets,
```

Comment:
I don't think this change is enough - note that the `get_store_resource` function, when not dealing with fset/fvec etc. actually does the following:
```
        stores = mlrun.store_manager.set(secrets, db=db)
        return stores.object(url=uri)
```
Which means that the secrets provided to the function will get "promoted" into store-secrets, and may serve for a scope much wider than this execution. I think you need to also add a `store_secrets` parameter to `get_store_resource` and inside it change the implementation to do:
```
return stores.object(url=uri, secrets=store_secrets)
```